apollo-router/src/protocols/websocket.rs

```diff
@@ -616,6 +745,26 @@ mod tests {
                     .await
                     .unwrap();
 
+                if let Some(duration) = heartbeat_interval {
+                   tokio::time::pause();
+                   assert!(
+                       socket.next().now_or_never().is_none(),
+                       "It should be no pending messages"
+                   );
+
+                   tokio::time::sleep(duration).await;
+                   let ping_message = socket.next().await.unwrap().unwrap();
+                   assert_eq!(ping_message, AxumWsMessage::Text(
+                       serde_json::to_string(&ClientMessage::Ping { payload: Some(serde_json_bytes::Value::String("APOLLO_ROUTER_HEARTBEAT".into())) }).unwrap(),
+                   ));
+
+                   assert!(
+                       socket.next().now_or_never().is_none(),
+                       "It should be no pending messages"
+                   );
+                   tokio::time::resume();
+                }
+
```

Comment:
> @IvanGoncharov Is there a test that exercises the new behaviour? As in it should check the heartbeat from the subgraph perspective.

Yes, it checked here. 
1. we stop timer
1. We check that there are no messages in a socket
1. advance timer for duration (60s)
1. check that we got ping
1. check that there are no messages in a socket
1. resume timer

It's done using a saved timer, so Tokio automatically advances the timer, and we don't need to actually wait for the 60s, so the test is fine.
I chose the 60s, so if the mock timer works for some reason, the test will timeout.